common/src/main/java/com/wynntils/services/mapdata/providers/builtin/PlayerProvider.java

```diff
@@ -0,0 +1,266 @@
+
+        public RemotePlayerLocation(HadesUser hadesUser) {
+            this.hadesUser = hadesUser;
+        }
+
+        @Override
+        public Location getLocation() {
+            return new Location((int) hadesUser.getX(), (int) hadesUser.getY(), (int) hadesUser.getZ());
+        }
+
+        @Override
+        public String getFeatureId() {
+            return hadesUser.getName().toLowerCase(Locale.ROOT);
+        }
+
+        @Override
+        public String getCategoryId() {
+            return "wynntils:players:" + (hadesUser.getRelation().isEmpty() ? "other" : hadesUser.getRelation());
+        }
+
+        @Override
+        public Optional<MapAttributes> getAttributes() {
+            return Optional.of(new AbstractMapAttributes() {
+                @Override
+                public Optional<String> getLabel() {
+                    return Optional.of(hadesUser.getName());
+                }
+
+                @Override
+                public Optional<CustomColor> getLabelColor() {
+                    return Optional.of(hadesUser.getRelationColor());
+                }
+
+                @Override
+                public Optional<String> getIconId() {
+                    return Optional.of("wynntils:icon:player:head");
+                }
+
+                @Override
+                public Optional<MapDecoration> getIconDecoration() {
+                    return Optional.of(new MainMapIconDecoration());
+                }
+            });
+        }
+
+        @Override
+        public List<String> getTags() {
+            return List.of();
+        }
+
+        private final class MainMapIconDecoration implements MapDecoration {
+            private static final float INITIAL_PLAYER_HEAD_RENDER_SIZE = 24;
+
+            @Override
+            public boolean isVisible() {
+                return true;
+            }
+
+            @Override
+            public void render(
+                    PoseStack poseStack,
+                    MultiBufferSource bufferSource,
+                    boolean hovered,
+                    boolean fullscreenMap,
+                    float zoomLevel) {
+                float playerHeadRenderSize = INITIAL_PLAYER_HEAD_RENDER_SIZE
+                        * (fullscreenMap
+                                ? 1
+                                : Managers.Feature.getFeatureInstance(MinimapFeature.class)
+                                        .minimapOverlay
+                                        .remotePlayersHeadScale
+                                        .get());
+
+                poseStack.pushPose();
+                // center the player icon
+                poseStack.translate(-playerHeadRenderSize / 2f, -playerHeadRenderSize / 2f, 0);
+
+                HadesUser user = RemotePlayerLocation.this.hadesUser;
+                ResourceLocation skin = SkinUtils.getSkin(user.getUuid());
+
+                if (!fullscreenMap) {
+                    // outline
```

Comment:
I've been thinking about this a bit, like could we give the user the option to control whether outline and health bar should be drawn or not? I toyed with the idea of passing custom attributes but I don't think that's a good idea.

If I said that we can split this up into three different providers, `wynntils:player:head:<relation>`,  `wynntils:player:outline:<relation>` and  `wynntils:player:health:<relation>`, what would you say? Is that a good idea? My thought was that you can then select if you want the outline, head and health bar individually, if we allow different category selections for the minimap and the main map.

I'm not sure this is a good idea, but it feels kind of okay to me right now at least.

Reply:
I think this concept will break very easily, when we are about to add changes like "iconScale". It'll also be confusing when hovering/clicking on the map, as it won't propagate down. I see what you want to achieve here, but I am against splitting this into pieces.

Reply:
Ok, your objections sound reasonable. I withdraw that idea. :)